src/libraries/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -517,14 +517,6 @@ private static unsafe bool ParseNumber(ref char* str, char* strEnd, NumberStyles
                             {
                                 exp = exp * 10 + (ch - '0');
                                 ch = ++p < strEnd ? *p : '\0';
-                                if (exp > 1000)
```

Comment:
Sorry for the delay in review here. 

This isn't quite the right fix and likely introduces a different bug. For example, if you have an exponent of `21474836492` it will succesfully parse the first `9` digits giving `214748364`. Multiply by 10 gives `2147483640`, add `9` overflows to `-2147483647`. Then further multiplied by 10, gives exactly `10`, adding `2` gives `12`.

Both of these last two digits are problematic because of this and so we need something that accounts for overflow here and surfaces it as invalid input/failure to parse.

Reply:
@tannergooding , I agree with you in this case. I have idea about processing this cases

Reply:
@tannergooding , I had just repeat my summer investigation about algorithms of BigInteger, and I remember, that there is [BigNumber](https://github.com/dotnet/runtime/blob/22b516ca0a4b9851cf993ecc83b2a8238898c465/src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs#L282) class with [BigNumberBuffer](https://github.com/dotnet/runtime/blob/22b516ca0a4b9851cf993ecc83b2a8238898c465/src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs#L292) inside it and this buffer contains [scale](https://github.com/dotnet/runtime/blob/22b516ca0a4b9851cf993ecc83b2a8238898c465/src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs#L296) with int type.
BigNumber provide some functionality to BigInteger and if int type with it's [MaxValue](https://docs.microsoft.com/en-us/dotnet/api/system.int32.maxvalue?view=net-5.0) is invalid, then .NET Command should create separate task for change it.
But firstly, lets make little calculation:
itemSize * arraySize = 32bit * 2147483647 = 68719476704 bit = 8589934588 byte ≈ 8388608 kbyte = 8192 mb = 8 gb
thus, theoretically we can provide 8 Gb for store one BigInteger by current architecture! I think, this is more than enough

Reply:
I think we can simply make this be:
```csharp
int exp = 0;
do
{
    exp = exp * 10 + (ch - '0');
    ch = ++p < strEnd ? *p : '\0';
} while ((ch >= '0') && (ch <= '9') && (exp < 214748364));

if ((ch >= '0') || (ch <= '9'))
{
    // We still had remaining characters but bailed early because
    // the exponent was going to overflow. If exp is exactly 214748364
    // then we can technically handle one more character being 0-7
    // but the additional complexity might not be worthwhile.

    Debug.Assert(exp >= 214748364);
    return false;
}
```

Reply:
The point is to account for overflow and ensure its handled so we can throw for `Parse` and return `false` for `TryParse`.

This trivially handles it by accounting for both overflow due to `exp * 10` (which can only overflow for `exp > (int.MaxValue / 10)`) and for the subsequent add which can only overflow for `exp == (int.MaxValue / 10)` and `ch > '7'`